library/packages/test/y2packager/licenses_fetchers/rpm_test.rb

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env rspec
+# ------------------------------------------------------------------------------
+# Copyright (c) 2018 SUSE LLC, All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+# ------------------------------------------------------------------------------
+
+require_relative "../../test_helper"
+require_relative "./shared_examples"
+
+require "y2packager/licenses_fetchers/rpm"
+require "y2packager/package"
+
+describe Y2Packager::LicensesFetchers::Rpm do
+  subject(:fetcher) { described_class.new(product_name) }
+
+  let(:lang) { "en_US" }
+  let(:product_name) { "SLES" }
+  let(:package_name) { "sles-release" }
+  let(:package_properties) { [{ "product_package" => package_name }] }
+  let(:package_status) { :selected }
+  let(:package) { instance_double(Y2Packager::Package, status: package_status, extract_to: nil) }
+
+  before do
+    allow(Yast::Pkg).to receive(:ResolvableProperties)
+      .with(product_name, :product, "")
+      .and_return(package_properties)
+
+    allow(Y2Packager::Package).to receive(:find)
+      .with(package_name)
+      .and_return([package])
+  end
+
+  it_behaves_like "a fetcher"
+
+  describe "#content" do
+    context "when a selected package is found" do
+      let(:license_path) { "/fake/path/to/#{lang}/LICENSE" }
+      let(:default_license_path) { "/fake/path/to/default/txt/license" }
+      let(:license_file_content) { "" }
+
+      before do
+        allow(Dir).to receive(:glob).and_call_original
+        allow(Dir).to receive(:glob).with(/#{lang}/, anything).and_return([license_path])
+        allow(Dir).to receive(:glob).with(/LICENSE.TXT/, anything).and_return([default_license_path])
+        allow(File).to receive(:read).with(license_path).and_return(license_file_content)
+        allow(File).to receive(:read).with(default_license_path).and_return(license_file_content)
```

Comment:
this huge amount of mocking a bit scare me. Is possible to just add to data section few examples rpms on which we do tests? This will also allow in future testing potentially not working rpms ( read it as "we change it and expect that yast will work" )

Reply:
I agree. I'll try to do what you are proposing.